src/main/groovy/com/readytalk/gradle/plugins/ci/CiInfoPlugin.groovy

```diff
@@ -0,0 +1,174 @@
+package com.readytalk.gradle.plugins.ci
+
+import com.readytalk.gradle.util.PluginUtils
+import com.readytalk.gradle.util.StringUtils
+import nebula.plugin.info.InfoBrokerPlugin
+import nebula.plugin.info.basic.BasicInfoPlugin
+import nebula.plugin.info.java.InfoJavaPlugin
+import nebula.plugin.info.reporting.InfoJarManifestPlugin
+import nebula.plugin.info.reporting.InfoJarPropertiesFilePlugin
+import nebula.plugin.info.reporting.InfoPropertiesFilePlugin
+import nebula.plugin.info.scm.ScmInfoPlugin
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.lib.RepositoryBuilder
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.publish.ivy.tasks.GenerateIvyDescriptor
+
+class CiInfoPlugin implements Plugin<Project>, PluginUtils {
```

Comment:
There was some major refactoring action in this class to make sure all the functions were doing one thing, reduce the size of the main apply() function, and make the code a bit more self-documenting and easy to understand.